### PR TITLE
pkgs/development/libraries: add missing descriptions and homepages

### PR DESCRIPTION
**Pull request by cko**
###### Motivation for this change 
Many packages below pkgs/development/libraries lack a description and sometimes also a homepage in their metadata. This PR adds some.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by c0bw3b** [comment]
From our [guidelines](https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes), short descriptions shouldn't end with a dot.

File: pkgs/development/libraries/ilmbase/default.nix [hunk 1]
```diff
@@ -40,6 +40,7 @@ stdenv.mkDerivation rec {
   doCheck = stdenv.isx86_64;
 
   meta = with stdenv.lib; {
+    description = "A basic, light-weight, and efficient representation of 2D and 3D vectors and matrices and other mathematical objects, functions, and data types common in computer graphics applications.";
```
Comment (c0bw3b) [inline pkgs/development/libraries/ilmbase/default.nix]: Probably too long.
Other distros' desc: https://repology.org/project/ilmbase/information

**Comment by cko** [comment]
> From our [guidelines](https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes), short descriptions shouldn't end with a dot.

oh, damn. I'm mixing this up every single time. Fixed now.

File: pkgs/development/libraries/ilmbase/default.nix (hunk 1 above)
Comment (cko) [reply pkgs/development/libraries/ilmbase/default.nix]: fixed

**Review by c0bw3b — APPROVED** [review]

**Pull request merged**